src/IO/Observer/GetSectionObservationKey.hpp

```diff
@@ -17,13 +18,15 @@ namespace observers {
 /// parallel algorithms. The return value can be used to construct a subfile
 /// path for observations, and to skip observations on elements that are not
 /// part of a section (`std::nullopt`).
-template <typename SectionIdTag, typename DbTagsList>
+template <typename SectionIdTag, typename... Params ,
+          template <class...> class U>
 std::optional<std::string> get_section_observation_key(
-    [[maybe_unused]] const db::DataBox<DbTagsList>& box) noexcept {
+    [[maybe_unused]] const U<Params...>& box) noexcept {
```

Comment:
How about `template <class...> class U` -> `typename Box`, and delete the `Params...` pack (it's not needed here)